src/compiler/reviews.md

```diff
@@ -1,55 +1,388 @@
+  isolation but are problematic or undesirable in the larger context.
+
+Reviewing accomplishes this by bringing in another set of eyes to look at a
+proposed change from a different perspective, which increases the chance of
+catching mistakes early and uncovering potential blind spots in the reasoning
+behind the change.
+
+
+## Basic Reviewing Requirements
+
+There are a number of requirements that need to be met in order for reviewing to
+be effective:
+
+- Reviewers must have a sufficient **understanding of the code** under review.
+  - This is important to help spot non-obvious, unintentional side effects of a
+    given change.
+- Pull request authors must provide:
+  1. A concise high-level **description of the change** and (2) the
+     **rationale** behind it, unless the change is extremely trivial, like typo
+     fixes.
+  2. For the rationale to be even more useful, authors are encouraged to list
+     potential **points of contention**, **compromises** that needed to be made,
+     **alternative approaches** that have been considered, **relevant
+     documentation, discussions and context**, etc.
+    - Reviewing code is difficult, and reviewers only have a **limited amount of
+      time** to do it. Jump-starting the review process by not making the
+      reviewer puzzle together the intention and context of a pull request will
+      not only speed things up but also improve the quality of the review.
+- Reviewers must have a good idea on whether they are the **right person to
+  approve** the change.
+  - Knowledge of the code under review is an obvious but not the only criteria
+    for answering this question.
+  - Procedure wise, the reviewer also needs to decide:
+    - Can the reviewer make the decision alone?
+    - Does the PR needs to go through the [Major Change Process][mcp], or does
+      it need a T-compiler [Final Comment Period][fcp] sign-off, or does it need
+      a full [Request For Comments (RFC)][rfc]?
+    - Does the PR need reviews and/or sign-off from other teams, particularly
+      T-lang?
+    - Can the changes break stable code or begin accepting new code that we do
+      not intend to? If the PR contains risks, is it sufficiently justified?
+      Does the changes need ecosystem impact evaluation through crater runs?
+    - Will the PR introduce significant perf changes? If there might be a perf
+      regression, is it justified? Does the PR need a perf run? 
+    - Can the reviewer perform the review sufficiently thorough and in a timely
+      fashion?
+    - Is the reviewer impartial enough to provide a sufficiently unbiased
+      perspective? E.g. due to co-authorship (sufficiently significant changes
+      to the PR made by the reviewer) or other conflicts-of-interest?
+
+[rfc]: https://github.com/rust-lang/rfcs
+
+## Reviewing Checklist
+
+The following list of questions will help reviewers and PR authors alike to
+bring PRs into good shape and meet the above criteria:
```

Comment:
I'm not sure how useful this list is when it solely exists in the forge.

I think at minimum we should change the PR template to include a link to it.

In my personal ideal world we'd have this list as a list of checkboxes (reformulated so it doesn't say "Does the PR need X", but "Checked whether the PR needs X and handled it if necessary"

Reply:
This was originally phrases to be included as a PR template yeah.